memcached.c

```diff
@@ -5458,8 +5551,51 @@ static void drive_machine(conn *c) {
                 stats.rejected_conns++;
                 STATS_UNLOCK();
             } else {
+#ifdef TLS
+                SSL *ssl = NULL;
+                if (IS_TCP(c->transport) && settings.ssl_enabled) {
+                    // Setup the SSL object at each connection level when it's enabled
+                    // at the correct port if ssl_port is specified.
+                    int port = 0;
+                    if (settings.ssl_port > 0) {
```

Comment:
Is this because SSL and non-SSL listeners can co-exist?

Reply:
Yes. Tests are in t/ssl_ports.t

Reply:
This seems confusing as implemented, or would be hard to explain to a user, maybe. I'll think about it more but it wasn't obvious to me and I wrote half this code :)

Reply:
Let me refactor the code, probably the previous version was more clear, and add a better comment.

Reply:
I mean from an end-user standpoint too.
./memcached -Z (--enable-ssl) -l 127.0.0.1:11211 -l 127.0.0.1:11212 -l 0.0.0.0:11311 -o ssl_port=11211

... means "secure any listener on just port 11211" - I'm not sure how someone would come to that conclusion by looking at the help output.

might be better to track tls-or-not on the c-> level and mark it in the listener object during setup.

I'm not sure I like this idea but something like: `-Z -l 0.0.0.0:11211 -l notls:127.0.0.1:11211

Then you're explicitly stating which listeners do what, and if TLS is enabled it's simply the default unless otherwise specified.

Reply:
So we implement this at the interface+port level, getting rid off the ssl_port. Let me give this a thought, sounds like this suggestion aligns with the original intention of this feature (do plain text over the localhost).